public/app/plugins/datasource/prometheus/configuration/AzureCredentialsForm.tsx

```diff
@@ -142,19 +165,17 @@ export const AzureCredentialsForm = (props: Props) => {
   };
 
   const onSubscriptionChange = (selected: SelectableValue<string> | undefined) => {
-    if (onCredentialsChange) {
-      const updated: AzureCredentials = {
-        ...credentials,
-        defaultSubscriptionId: selected?.value,
-      };
-      onCredentialsChange(updated);
-    }
+    const updated: AzureCredentials = {
+      ...credentials,
+      defaultSubscriptionId: selected?.value,
+    };
+    onCredentialsChange(updated);
   };
   const prometheusConfigOverhaulAuth = config.featureToggles.prometheusConfigOverhaulAuth;
 
   return (
     <div className="gf-form-group">
-      {props.managedIdentityEnabled && (
+      {(managedIdentityEnabled || workloadIdentityEnabled) && (
```

Comment:
nit:  I think ADX has cleaner implementation ([here](https://github.com/grafana/azure-data-explorer-datasource/blob/8ba70e1819498e00bd91282a6795089d160002ba/src/components/ConfigEditor/AzureCredentialsForm.tsx#L131C38-L131C38)). It's OK either way but if we're going to move it to shared library, I would rather pick that implementation.

Reply:
I agree the ADX implementation is cleaner, when we move to the shared library that is the implementation we will use.